src/odbc_command_handler.rs

```diff
@@ -73,15 +72,17 @@ impl OdbcCommandHandler {
         req: GetCommandSchemaRequest,
     ) -> Result<(), MyServerError> {
         let connection = self.get_connection()?;
-        let cursor = self.get_result_cursor(&connection, req.command.clone())?;
+        let mut cursor = self.get_result_cursor(&connection, req.command.clone())?;
         let ticket = req.command.to_ticket();
-        self.send_schema_from_cursor(req.response_sender, cursor, ticket)
+        self.send_schema_from_cursor(req.response_sender, &mut cursor, ticket)
     }
 
     fn handle_get_command_data(&mut self, req: GetCommandDataRequest) -> Result<(), MyServerError> {
         let connection = self.get_connection()?;
-        let cursor = self.get_result_cursor(&connection, req.command.clone())?;
-        self.send_flight_data_from_cursor(req.response_sender, cursor)
+        let mut cursor = self.get_result_cursor(&connection, req.command.clone())?;
+        let schema = arrow_odbc::arrow_schema_from(&mut cursor)?;
+        let unmut_cursor = cursor;
```

Comment:
Why do you need to rebind cursor?

Reply:
To get a schema, it needs a mutable cursor. But the following function call needs a unmutable cursor.